src/main/java/bio/terra/stairway/SafeHashMap.java

```diff
@@ -0,0 +1,98 @@
+package bio.terra.stairway;
+
+import bio.terra.stairway.exception.DatabaseOperationException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * SafeHashMap wraps a HashMap<String, Object>
+ * It provides a subset of the HashMap methods. It localizes code that casts from Object to
+ * the target type. It provides a way to set the map to be immutable.
+ */
+public class SafeHashMap {
+    private Map<String, Object> map;
+    private ObjectMapper objectMapper;
+
+    public SafeHashMap() {
+        map = new HashMap<>();
+    }
+
+    /**
+     * Convert the map to an unmodifiable form.
+     */
+    public void setImmutable() {
+        map = Collections.unmodifiableMap(map);
+    }
+
+    /**
+     * Return the object from the hash map cast to the right type.
+     * Return null, if the Object cannot be cast to that type.
+     *
+     * @param key - key to lookup in the hash map
+     * @param type - class requested
+     * @return null if not found
+     * @throws ClassCastException if found, not castable to the requested type
+     */
+    public <T> T get(String key, Class<T> type) {
+        Object value = map.get(key);
+        if (value == null) {
+            return null;
+        }
+
+        if (type.isInstance(value)) {
+            return type.cast(value);
+        }
+        throw new ClassCastException("Found value '" + value.toString() +
+                "' is not an instance of type " + type.getName());
+    }
+
+    public void put(String key, Object value) {
+        map.put(key, value);
+    }
+
+    // TODO: Add other methods as they are needed
+
+
+    public String toJson() {
```

Comment:
I'm in favor of methods that help us convert to from Json but I'm not sure they belong in this class. I don't think we need a SafeHashMap - we should use Collections.synchronizedMap or Collections.immutableMap as needed and then have a utility class that we can use anywhere for conversions if needed.

Reply:
We should discuss. SafeHashMap is a really bad name for what that is. I don't mean for it to be general-purpose. There are three things I wanted it to do:

1. Support the lifecycle of the flight parameters that start out modifiable and then get set immutable when the flight is submitted.
1. Wrap the required casting into a templated method so we don't have casts sprinkled all over the code.
1. Wrap the to/from Json and generate the right exception for stairway.

I think of it as a utility class. Can you clarify what you would have me replace that with?

Reply:
Renamed the class to FlightMap so it is clear it is custom for Stairway and not a general-purpose thing.